table/namespace_classifier.go

```diff
@@ -158,6 +158,10 @@ func (c *tableNamespaceClassifier) GetRegionNamespace(regionInfo *core.RegionInf
 	return namespace.DefaultNamespace
 }
 
+func (c *tableNamespaceClassifier) AllowMerge(one *core.RegionInfo, other *core.RegionInfo) bool {
+	return Key(one.StartKey).TableID() == Key(other.StartKey).TableID()
```

Comment:
how about the metadata which has no TableID?

Reply:
if the key is not table key, TableID() will return 0